### DISCUSSION TITLE
Limit rows returned from select based on the type of the user

### DISCUSSION THREAD

**Discussion by stargazer-ad** [supabase/supabase#28469] [Questions]
Hi supabase team, 

I am facing an issue which I am not able to model into RLS. Here is the background. I have a dataset table that stores all the researched data. I want to enforce the following guidelines.

1) A user on the 'free' plan should not be able to query more than 5 rows.
2) Limit access from the api from the frontend to disallow a user to export the raw data. I am currenly able to make a curl with the browser rest api and export all the data by doing a select *

What are the ways this can be accomplished?

**Comment by GaryAustin1** [comment]
A view or rpc call to a postgres function that checks the user plan and limits the data.
Otherwise a server side call on your part that does the check and limit.

You can't limit from things like Postman as they look the same to the REST API and almost all headers can be faked.  The JWT is the only unique factor and RLS  can't limit rows returned as it is a per row operation.
